.github/workflows/update.yml

```diff
@@ -113,6 +121,38 @@ jobs:
             echo "EOF" >> "$GITHUB_OUTPUT"
           fi
 
+      - name: Update dev flake.lock
+        id: dev_flake_lock
+        if: inputs.dev_lock || github.event_name == 'schedule'
+        run: |
+          root_nixpkgs=$(nix eval -f . 'inputs.nixpkgs.rev')
+          old=$(git show --no-patch --format=%h)
+          nix flake update --commit-lock-file \
+              --override-input nixpkgs "github:NixOS/nixpkgs/$root_nixpkgs" \
+              --flake './flake/dev'
+          new=$(git show --no-patch --format=%h)
+          if [ "$old" != "$new" ]; then
+            echo "body<<EOF" >> "$GITHUB_OUTPUT"
+            git show --no-patch --format=%b >> "$GITHUB_OUTPUT"
+            echo "EOF" >> "$GITHUB_OUTPUT"
+          fi
```

Comment:
We essentially copy-paste this workflow step three times. I wonder if there's a better approach? Maybe we can [avoid duplication](https://docs.github.com/en/actions/sharing-automations/avoiding-duplication) by adding a composite action? I think that'd be best in a follow-up PR as I like to keep workflow changes to a minimum per-PR